File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PatchExperimentExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PatchExperimentExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var experimentPatchInput = new ExperimentPatchInput(
            instructions: JsonSerializer.Deserialize<List<Dictionary<string, object>>>("""
                [
                    {
                        "kind": "updateName",
                        "value": "Updated experiment name"
                    }
                ]
            """),
            comment: "Example comment describing the update"
        );

        try
        {
            var response = new ExperimentsApi(config).PatchExperiment(
                projectKey: "projectKey_string",
                environmentKey: "environmentKey_string",
                experimentKey: "experimentKey_string",
                experimentPatchInput: experimentPatchInput
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling ExperimentsApi#PatchExperiment: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PatchExperimentExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const experimentPatchInput: api.ExperimentPatchInput = {
  instructions: [
    {
      "kind": "updateName",
      "value": "Updated experiment name"
    }
  ],
  comment: "Example comment describing the update",
};

new api.ExperimentsApi(configuration).patchExperiment(
  "projectKey_string", // projectKey
  "environmentKey_string", // environmentKey
  "experimentKey_string", // experimentKey
  experimentPatchInput,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling ExperimentsApi#patchExperiment:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
